geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/AttributeIdxStrategy.scala

```diff
@@ -331,6 +332,18 @@ object AttributeIndexStrategy extends StrategyProvider {
         checkOrder(f.getExpression1, f.getExpression2).filter(indexed)
             .map(p => StrategyDecision(new AttributeIdxRangeStrategy, costp(p)))
 
+      // not check - we only support 'not null' for an indexed attribute
+      case n: Not =>
+        Option(n.getFilter).collect { case f: PropertyIsNull => f }.flatMap { f =>
```

Comment:
Yeah, it's so we can collect on the filter we want - it could have alternatively been written as a if(.isInstanceOf[PropertyIsNull]), but that seemed uglier...